### ISSUE TITLE
Drain hosted channel fails repeatedly

### ISSUE THREAD

**Issue by nanotube** [btcontract/lnwallet#19]
When trying to drain hosted channel, the process fails repeatedly, leaving a 'lightning payment expired and cancelled' rebalancing transaction in the list.
I have the day earlier successfully made a couple of rebalancing payments, but now after several days of trying, it keeps failing. 
perhaps it keeps trying to use the same loop route and there's a node in the route that fails.... if so BLW should try different routes after several failures.
I have 9 other channel besides the hosted one, so there should be plenty of route possibilities to try.
I'd be glad to post transaction debug info or any other info you may request.

**Comment by btcontract** [comment]
Yes, please provide a debug info for a failed payment, that could help.

**Comment by nanotube** [comment]
Hi, here's the debug info, below.
Also, I notice the 'payee node id' is not the same as my node id...

=============
Your wallet
-->
Node ID: 03144fcc73cea41a002b2865f98190ab90e4ff58a2ce24d3870f5079081e42922d, Channel ID: 608205x2833x0, Expiry: 144 blocks, Routing fee: 0.010000% of payment sum + baseline 1000 msat
-->
Node ID: 03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f, Channel ID: 619818x3023x1, Expiry: 144 blocks, Routing fee: 0.010000% of payment sum + baseline 1000 msat
-->
Node ID: 03c436af41160a355fc1ed230a64f6a64bcbd2ae50f12171d1318f9782602be601, Channel ID: 616299x1652x0, Expiry: 144 blocks, Routing fee: 0.000200% of payment sum + baseline 2 msat
-->
Payee node ID: 0300b695e7f9e7bbf323b359d9113f69ee7af60486e0f9669dcdaccc15b5371cd9, Expiry: 303 blocks

DecryptedFailurePacket(03144fcc73cea41a002b2865f98190ab90e4ff58a2ce24d3870f5079081e42922d,TemporaryChannelFailure(ChannelUpdate(ByteVector(71 bytes, 0x304402205ace2372a3fbeab53d66c6d6762b8c1b2347b549910c7e5ba36f20a84acafb8c022029fad8a1a5003b3a7b096e5ace358a27920871a56c34106ae0916ed81df333ff01),ByteVector(32 bytes, 0x6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000),684074353403822080,1588042414,1,1,144,1000,1000,100,Some(4000000000),ByteVector(empty))))
==
DecryptedFailurePacket(03144fcc73cea41a002b2865f98190ab90e4ff58a2ce24d3870f5079081e42922d,TemporaryChannelFailure(ChannelUpdate(ByteVector(71 bytes, 0x304402203e6a003ca402c775e73eb7dd08c2166a98b7d1e8cdd6c92e92767379570e735e02204e5b44ceef79fb2b4c66ce6514e31fe4bd2e191337500438da80763e82f6dcec01),ByteVector(32 bytes, 0x6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000),684563636071104512,1587799669,1,1,144,1,1000,100,Some(1100000000),ByteVector(empty))))

**Comment by btcontract** [comment]
> Also, I notice the 'payee node id' is not the same as my node id...

This is normal and immaterial to issue at hand, BLW obfuscates payee node id for some time now.

Both payment attempts return `TemporaryChannelFailure` error from my node which means those two attempted channels are incapable of routing at this time. 

Overall the connectivity between `03144fcc73cea41a002b2865f98190ab90e4ff58a2ce24d3870f5079081e42922d` and `03c436af41160a355fc1ed230a64f6a64bcbd2ae50f12171d1318f9782602be601` is low since 

```
./eclair-cli findroutetonode --nodeId=03c436af41160a355fc1ed230a64f6a64bcbd2ae50f12171d1318f9782602be601 --amountMsat=10000`
```

returns a single route:

```
[
  "03144fcc73cea41a002b2865f98190ab90e4ff58a2ce24d3870f5079081e42922d",
  "02f3069a342ae2883a6f29e275f06f28a56a6ea2e2d96f5888a3266444dcf542b6",
  "03c436af41160a355fc1ed230a64f6a64bcbd2ae50f12171d1318f9782602be601"
]
```

So it seems like this is not a bug, just a state of affairs.

**Comment by nanotube** [comment]
thank you for looking into it!
Some questions/thoughts:

since this is for draining the hosted channel, i presume the starting node, 03144fcc73cea41a002b2865f98190ab90e4ff58a2ce24d3870f5079081e42922d, is the BLW node? Perhaps BLW could use to open some more channels to better connect its hosted node to the network?

My BLW wallet has 6 other channels with incoming capacity besides the hosted channel, but it appears to always choose the drain amount that's equal to the incoming capacity on only one of them. Perhaps BLW could rotate through possible incoming channels when trying to drain, rather than keep trying the same one?

Thank you again~

**Comment by btcontract** [comment]
This could actually be a bug fixed in https://github.com/btcontract/lnwallet/releases/tag/0.4.5, would appreciate if you retry with v0.4.5

**Comment by nanotube** [comment]
Hi,
Thank you for following up on this.
Draining actually started working a few days after we last spoke on this issue. And I have successfully conducted quite a few drain transactions since then. Currently on version 0.4.4-150.
